content/post/gld.md

```diff
@@ -0,0 +1,1363 @@
+to find them using the excellent [jdkato/prose](https://github.com/jdkato/prose) NLP library for Go.
+
+Unfortunately, the devil is in the details. There are many unexpected variations for which we had to modify
+our initially straight-forward algorithm.
+
+## File names
+
+There are many possible license file names. A few examples:
+
+```nohighlight
+LICENSE
+license.txt
+License.md
+lisence.html
+lisense.rst
+copyright
+copying
+MIT.txt
+gpl-2.0
+legal
+```
+
+All of them may contain useful information, so we have to design a suitable regular expression for
+them, which is currently
+
+```go
+var (
+	licenseFileNames = []string{
+		"li[cs]en[cs]e(s?)",
+		"legal",
+		"copy(left|right|ing)",
+		"unlicense",
+		"l?gpl([-_ v]?)(\\d\\.?\\d)?",
+		"bsd",
+		"mit",
+		"apache",
+	}
+	licenseFileRe = regexp.MustCompile(
+		fmt.Sprintf("^(|.*[-_. ])(%s)(|[-_. ].*)$",
+			strings.Join(licenseFileNames, "|")))
+)
+```
+
+There may also be directories which are named like a license file, and we need to look inside.
+A few projects contained symbolic links to the actual license texts, and we need to resolve them.
+One project even has a license file which consists of the path to a real `LICENSE` file with a custom file name - we treat those as symlinks.
+
+## Rendering and normalization
+
+Many developers like when the licenses are displayed on GitHub nicely formatted, either
+put directly in HTML or written using a markup language such as Markdown or ReStructuredText.
+Reading those files verbatim is harmful for our matching core and decreases the detection rate
+dramatically. Thus we should first render markup to HTML and then extract plain text content from
+HTML. go-license-detector currently supports Markdown through
+[russross/blackfriday](https://github.com/russross/blackfriday) and ReST through
+[hhatto/gorst](https://github.com/hhatto/gorst). HTML tags are stripped with `golang.org/x/net/html`
+and a custom HTML entity recognizer.
+
+Having a plain text license, we need to normalize it. SPDX has a list of rules which do not affect
+accuracy, and we leverage it. However, our goal is data mining, so we can normalize aggressively.
+We designed a three-level normalization pipeline. The first one is SPDX with some other rules
+which do not affect the detection accuracy. The second one removes punctuation and lines with
+copyright information. We apparently lose some data but our detection is more robust to random
+deviations such as dots in the end of the section names or multiple copyright notices in the
+header. Finally, the third level removes letter accents (e.g. ñ becomes n, á becomes a, etc.) and
+removes all non-alphanumeric characters.
+
+To summarize, this is the evolution of the license file content inside go-license-detector's normalizer.
+
+Original ([home-assistant/home-assistant](https://raw.githubusercontent.com/home-assistant/home-assistant/dev/LICENSE.md)):
+{{% scroll-panel height="400" %}}
+```markdown
```

Comment:
I'd consider using the new `code` shortcode to embed the code, rather than having it right here.